src/shared/clients/codecatalystClient.ts

```diff
@@ -39,6 +40,13 @@ export interface CodeCatalystConfig {
     readonly gitHostname: string
 }
 
+const configToEnvMap: { [K in keyof CodeCatalystConfig]: string } = {
+    region: '__CODECATALYST_REGION',
+    endpoint: '__CODECATALYST_ENDPOINT',
+    hostname: '__CODECATALYST_HOSTNAME',
+    gitHostname: '__CODECATALYST_GIT_HOSTNAME',
+}
```

Comment:
I wonder if it makes sense to generalize the env var override pattern as part of the work done in https://github.com/aws/aws-toolkit-vscode/pull/4324

Just a future idea, doesn't need to be solved here.